bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/overheads/Overheads.scala

```diff
@@ -0,0 +1,130 @@
+package edu.gemini.model.p1.overheads
+
+import java.time.Duration
+
+import edu.gemini.model.p1.immutable._
+
+import scalaz.Scalaz._
+
+// This is not the ideal package for this to live in, but to avoid circular bundle references, we put it here.
+sealed trait Overheads {
+  def partnerOverheadFraction: Double
+  def acquisitionOverhead: Duration
+  def typicalExpTime: Option[Duration]
+  def otherOverheadFraction: Double
+
+  // Careful here: if we convert to hours directly using java.time API, this will be a Long value representing
+  // the floor of the number of hours.
+  private def acqOverheadHrs = acquisitionOverhead.toMinutes.toDouble / 60.0
+
+  def calculate(intTime: TimeAmount): ObservationTimes = {
+    // Calculate everything in hours, as per REL-2985 formulae.
+    val progTimeHrs  = {
+      val visTimeHrs = Overheads.visTimeHrs
+      val intTimeHrs = intTime.toHours.value
+      val overheadTimeHrs = (1 + otherOverheadFraction) * intTimeHrs
+      val numAcqs = ((overheadTimeHrs / visTimeHrs).toInt + ((overheadTimeHrs % visTimeHrs > 0) ? 1 | 0)).toDouble
```

Comment:
I'll have to ask on the REL task. My impression is that it will always be fixed to 2, but as it comes up in multiple places in the calculation, I extracted it.

The whole concept of floating point modulo arithmetic seems very strange to me. I understand what it is doing, but coming from a math background, I always associated it with number theory and the ring of integers, but I can see how it works.